src/solver/eigen/eigen_solver_factory.cpp

```diff
@@ -53,11 +53,21 @@ std::unique_ptr<eigen_solver> make_eigen_solver(json const& solver_data)
 
     if (std::string const& type = solver_data["type"]; type == "power_iteration")
     {
+#ifndef ENABLE_OPENCL
```

Comment:
These would need to specify what option requires OPENCL support (i.e. The "lanczos" option...).